src/Resource/Detectors/K8s.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Solarwinds\ApmPhp\Resource\Detectors;
+
+use Exception;
+use OpenTelemetry\API\Behavior\LogsMessagesTrait;
+use OpenTelemetry\SDK\Common\Attribute\Attributes;
+use OpenTelemetry\SDK\Common\Configuration\Configuration;
+use OpenTelemetry\SDK\Resource\ResourceDetectorInterface;
+use OpenTelemetry\SDK\Resource\ResourceInfo;
+use OpenTelemetry\SDK\Resource\ResourceInfoFactory;
+use OpenTelemetry\SemConv\ResourceAttributes;
+use Solarwinds\ApmPhp\Common\Configuration\Variables;
+
+final class K8s implements ResourceDetectorInterface
+{
+    use LogsMessagesTrait;
+
+    private const NAMESPACE_FILE_WINDOWS = 'C:\\var\\run\\secrets\\kubernetes.io\\serviceaccount\\namespace';
+    private const NAMESPACE_FILE_LINUX = '/run/secrets/kubernetes.io/serviceaccount/namespace';
+    private const MOUNTINFO_FILE = '/proc/self/mountinfo';
+    private const UID_REGEX = '/[0-9a-f]{8}-(?:[0-9a-f]{4}-){3}[0-9a-f]{12}/i';
+    private string $namespaceFile;
+    private string $mountInfoFile;
+
+    public function __construct(
+        ?string $namespaceFile = null,
+        ?string $mountInfoFile = null,
+    ) {
+        $this->namespaceFile = $namespaceFile ?? (PHP_OS_FAMILY === 'Windows' ? self::NAMESPACE_FILE_WINDOWS : self::NAMESPACE_FILE_LINUX);
+        $this->mountInfoFile = $mountInfoFile ?? self::MOUNTINFO_FILE;
+    }
+
+    public function getResource(): ResourceInfo
```

Comment:
Could we start adding some in-code documentation as we add more functions? For example, there are cases like `test_prefers_env_over_files` or returning empty resource, so it would be good know early.

Reply:
Sure. Added a comment about the namespace required for k8s resource attributes.

We prefer env over files because we want to use the value from k8s Downward API (from yaml file) first. https://swicloud.atlassian.net/wiki/spaces/NIT/pages/3549954601/Proposal+on+cgroup+v2+container+ID+problem#K8s-downward-API

Reply:
Can we add something like this, to this function and others, for future developers?

```suggestion
    /**
     * Returns resource attributes related to the current Kubernetes pod such as namespace, UID, and name.
     *
     * @return ResourceInfo The resource information for the current pod, or an empty resource if the k8s namespace is not available.
     */
    public function getResource(): ResourceInfo
```